libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/tasks/Tasks.kt

```diff
@@ -844,16 +844,18 @@ abstract class KotlinCompile @Inject constructor(
             )
             when {
                 !inputChanges.isIncremental -> NotAvailableForNonIncrementalRun(classpathSnapshotFiles)
-                inputChanges.getFileChanges(classpathSnapshotProperties.classpathSnapshot).none() -> NoChanges(classpathSnapshotFiles)
+                // When inputChanges.isIncremental == true, we want to compile incrementally. However, if the incremental state (e.g. lookup
+                // caches or previous classpath snapshot) is missing, we need to compile non-incrementally. That could happen if in the
+                // previous run:
+                //   - Compilation happened using Kotlin daemon but with IC disabled (`kotlin.incremental=false`)
```

Comment:
This scenario is not possible as `isIncrementalCompilationEnabled` is an input.

Reply:
Ah right, I updated this code comment.